File: eio2024-ev/2024-12-08-ev/kook/solution/sol2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> p(n), m(n);
    for (int i = 0; i < n; ++i) {
        cin >> p[i];
    }
    for (int i = 0; i < n; ++i) {
        cin >> m[i];
    }

    bool voimalik = 1;

    for (int i = 1; i < n; ++i) {
        if (m[i] * p[i-1] != m[i-1] * p[i]) {
            voimalik = 0;
            break;
        }
    }

    if (voimalik) {
        cout << "JAH\n";
        for (int i = 1; i <= n; ++i) {
            cout << i << " \n"[i==n];
        }
    }
    else {
        cout << "EI\n";
    }

    return 0;
}

File: eio2024-ev/2024-12-08-ev/kook/solution/sol2.py
n = int(input())
p = list(map(int, input().split()))
m = list(map(int, input().split()))

voimalik = True
for i in range(1,n):
    if (m[i]*p[i-1] != m[i-1]*p[i]):
        voimalik = False
    
if (voimalik):
    print('JAH')
    print(*range(1,n+1))

else:
    print('EI')